frontend/pages/queries/edit/components/EditQueryForm/EditQueryForm.tsx

```diff
@@ -823,12 +844,11 @@ const QueryForm = ({
             isLoadingAppConfig={isLoadingAppConfig}
           />
         )}
-        {showSaveChangesModal && (
-          <SaveChangesModal
+        {showConfirmSaveChangesModal && (
+          <ConfirmSaveChangesModal
             onSaveChanges={promptSaveQuery()}
-            toggleSaveChangesModal={toggleSaveChangesModal}
             isUpdating={isQueryUpdating}
-            sqlUpdated={confirmSqlChange()}
+            {...{ toggleConfirmSaveChangesModal, showChangedSQLCopy }}
```

Comment:
Is this ... necessary? Aren't these just props that can be set directly?

Reply:
Instead of combining the spread props pattern and the classic props pattern in cases like this, my preference would be to either put all props into the spread object or use the classic pattern. That way they're all at the same level the reader doesn't have to worry about getting lost in the different levels of nested curly braces.